File: Kalorie/Kalorie/Core/NutritionLabelRecognition/RecognizedTextLine.swift
//
//  RecognizedTextLine.swift
//  Kalorie
//
//  Created by Josef Antoni on 13.09.2026.
//

import CoreGraphics

struct RecognizedTextLine: Codable, Equatable {

    // MARK: - Properties

    let text: String
    let boundingBox: CGRect
}

File: Android/app/src/main/java/antoni/kalorie/core/nutritionlabelrecognition/RecognizedTextLine.kt
package antoni.kalorie.core.nutritionlabelrecognition

data class NormalizedRect(
    val x: Double,
    val y: Double,
    val width: Double,
    val height: Double,
) {

    // MARK: - Properties

    val minX: Double get() = x
    val maxX: Double get() = x + width
    val midX: Double get() = x + width / 2
    val minY: Double get() = y
    val maxY: Double get() = y + height
    val midY: Double get() = y + height / 2
}

data class RecognizedTextLine(
    val text: String,
    val boundingBox: NormalizedRect,
)
